src/main/java/org/jabref/logic/importer/AuthorListParser.java

```diff
@@ -94,9 +95,7 @@ private static StringBuilder buildWithAffix(Collection<Integer> indexArray, List
      * @param listOfNames the String containing the person names to be parsed
      * @return a parsed list of persons
      */
-    public AuthorList parse(String listOfNames) {
-        Objects.requireNonNull(listOfNames);
-
+    public AuthorList parse(@NonNull String listOfNames) {
```

Comment:
As I told you before I am not a fan of this NonNull stuff, It servers the same purpose as the Objects.requiresNonNull

Reply:
Yeah, I know. We always have that debate :)

I'll collect pros and cons in an ADR.

TL;DR for me. It enables static code analysis for nulls - not some runtime exception stuff. JabRef should not die because of NPEs. 

Links - it was approved twice and did not cause any issues yet

- https://github.com/JabRef/jabref/pull/10840
- https://github.com/JabRef/jabref/pull/10521#issuecomment-1867142747